pkg/plugin/v3/api.go

```diff
@@ -196,6 +206,12 @@ func (p *createAPIPlugin) GetScaffolder() (scaffold.Scaffolder, error) {
 		return nil, fmt.Errorf("unknown pattern %q", p.pattern)
 	}
 
+	// Add core resource versions to the config if they are not the defaults so we don't unnecessarily
+	// clutter the resulting PROJECT file.
+	if p.crdVersion != defaultCRDVersion {
+		p.config.AddCoreResourceVersions(config.CRDResourceName, []string{p.crdVersion})
+	}
```

Comment:
IMO the version should be set in the resource as an attribute of it:  (not that it will make the implementation simplified too)

```
resources:
- group: cache
  kind: Memcached
  version: v1alpha1
  apiVersion: v1 
```

Then, here we would have a func whcih is very similar to : https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/model/config/config.go#L115-L126 to check if a resource was already scaffolded with an apiVersion which is NOT equals the apiVersion informed. 

See that if we do NOT respect the model domains it means that we are unable to address many uses cases described in https://github.com/kubernetes-sigs/kubebuilder/pull/1644#discussion_r505905571 after without a breaking change.  The K8S APIs will still be deprecating in the future as well so, it is not a one-time thing.

Reply:
👍